pkgs/development/tools/misc/binutils/default.nix

```diff
@@ -1,18 +1,23 @@
-{ stdenv, buildPackages
+{ stdenv, lib, buildPackages
 , fetchurl, zlib, autoreconfHook264
+# Enabling all targets increases output size to a multiple.
+, withAllTargets ? false, libbfd, libopcodes
+, enableShared ? true
 , noSysDirs, gold ? true, bison ? null
 }:
 
 let
+  reuseLibs = enableShared && withAllTargets;
```

Comment:
Woo! I am a fan of that.

Reply:
I'd advocate making it unconditional if only there was a way to prevent it from being built at all, sigh.

Reply:
Apart from complicating stdenv bootstrap, they're significantly larger due to supporting all targets.

Reply:
So far the design is that you have "simple" versions of the libs inside the binutils derivation, and then those more "featureful" variants built separately for use by other packages.

Reply:
Oh right, good point.